x/logic/prolog/context.go

```diff
@@ -18,5 +18,5 @@ func UnwrapSDKContext(ctx context.Context, env *engine.Env) (sdk.Context, error)
 		return sdkCtx, nil
 	}
 
-	return sdk.Context{}, ResourceError(ResourceContext(), env)
+	return sdk.Context{}, engine.ResourceError(ResourceContext(), env)
```

Comment:
Nope, I can't. `engine.ResourceError` is the `resource_error/1` well known predicate, in the form `resource_error(+Resource)` where `Resource` is the name of the resource, for instance `memory`, `max_threads', `stack_overflow`...

But your comment is pertinent. However, I prefer not to convey the error cause via an error message. I'm rather thinking about implementing more granular errors for resources, such as `context_absent` or `context_not_found`.

I suggest creating a ticket to address this later. It's not a priority because this type of error is unlikely to occur (by design, but knowing Murphy's law...).

Reply:
That would be great! Please do so.